src/Database/LSMTree/Internal/RawBytes.hs

```diff
@@ -218,6 +220,19 @@ instance Monoid RawBytes where
 copy :: RawBytes -> RawBytes
 copy (RawBytes pvec) = RawBytes (PV.force pvec)
 
+-- | Force 'RawBytes' to not retain any extra memory. This may copy the contents.
+force :: RawBytes -> ByteArray
+force (RawBytes (PV.Vector off len ba))
+    | off == 0
+    , BA.sizeofByteArray ba == len
+    = ba
```

Comment:
Is there still a good reason to use `copy` over this? I don't see when we would require a byte array to be copied if it doesn't retain unnecessary memory, but I guess we could avoid the branch in situations where we already know we're dealing with a slice.

If there are reasons, it would be nice to have a comment explaining when to use which. If not, you could just replace `copy`.

Reply:
Okay, let's keep them both.